fedmsg_meta_fedora_infrastructure/greenwave.py

```diff
@@ -32,17 +32,17 @@ class GreenwaveProcessor(BaseProcessor):
 
     @staticmethod
     def satisfied(msg):
-        try:
-            # https://pagure.io/greenwave/pull-request/100
-            return msg['msg']['policies_satisfied']
-        except KeyError:
-            return msg['msg']['policies_satisified']
+        return msg['msg']['policies_satisfied']
 
     def link(self, msg, **config):
         subject = msg['msg']['subject']
         if subject:
             base = "https://taskotron.fedoraproject.org/resultsdb/results"
-            query = urlencode(msg['msg']['subject'][0])
+            # we want to be sure that we always have the same parameter order
+            query = urlencode(sorted(
+                [(k, v) for k, v in msg['msg']['subject'][0].items()],
```

Comment:
I have been told that the `subject` might be more complicated then a simple key-value.
So we'll need to make sure that deeper nested subjects are also correctly encoded probably.

Reply:
Really?

Reply:
@ralphbean I can't find the exact quote, but it was strongly alluded to that that was one of the main reasons none of my alternative suggestions could be used in https://pagure.io/waiverdb/issue/134#comment-496071 .

I think I remember having a discussion on IRC later about that where people told me that the "subject" for things can be pretty much anything and that "subjects from the past don't give guarantees for the future", and that that's why subjects need to be encoded as json throughout the stack.

Unless waiverdb and greenwave both have different views on what a "subject" is?